pandas/core/internals/managers.py

```diff
@@ -1075,6 +1075,12 @@ def value_getitem(placement):
         blknos = self._blknos[loc]
         blklocs = self._blklocs[loc].copy()
 
+        if not self.axes[0].is_unique:
```

Comment:
really don't want to add *anything* in internals; why is this not possible to be done in pandas/core/indexing?

Reply:
@jreback agreed, thanks for feedback! This is work in progress. I'll try to fix it in pandas/core/indexing.